### ISSUE TITLE
AIS CPA and TCPA calculation seem wrong

### ISSUE THREAD

**Issue by mcdonaldajr** [SignalK/signalk-derived-data#48]
derived data plug in calculating CPA as 14,588 and TCPA as 40,956.
OpenCPN and SeaClear agree on CPA as 3.74NM and TCPA as 2hrs 48 minutes which I calculate at 6926 meters and 10,080 seconds.
Have I misunderstood the units?
Attached are the screenshots.

![Screenshot 2020-05-01 at 19 20 22](https://user-images.githubusercontent.com/4070175/80831195-3964d280-8be2-11ea-8b27-46c38b906a69.png)
![Screenshot 2020-05-01 at 19 21 17](https://user-images.githubusercontent.com/4070175/80831208-3e298680-8be2-11ea-82f4-0404e50572d0.png)

**Comment by sbender9** [comment]
@joabakk do you have time to look into this?

**Comment by mcdonaldajr** [comment]
Here's the AIS sample data I was using to test. I also fed simulated position of my boat 23.47E, 37.07N to 23.56E, 37.21N moving at 0.1 knots
Thanks for investigating. Anthony

[nmea_data_sample.txt](https://github.com/SignalK/signalk-derived-data/files/4567946/nmea_data_sample.txt)

**Comment by joabakk** [comment]
@sbender9 I will have a look

**Comment by joabakk** [comment]
@mcdonaldajr could you provide a sample file with your own gps positions as well? It is not so easy to simulate your boats movements and hit the calculations you were seeing. I think I found a bug, just need to be sure before I push it

**Comment by joabakk** [comment]
Hehe just spent an hour comparing the code with https://github.com/jaffadog/marine-cpa-tcpa/blob/master/app.js which it turns out, is copied from my code...

**Comment by mcdonaldajr** [comment]
Unfortunately, I simulated my GPA position using my laptop which has a GPS simulator on it, so I don't have a combined file. What I have done since then is use the CPN plugin ShipDriver as a source of GPS data. The only other thing I needed to do was use PathMapper signal K plugin to map navigation.speedThroughWater to navigation.speedOverGround as ShipDriver only simulates speedThoughWater. It is a shame that we can't slow down the reading from the AIS file to simulate real data coming in, as it just repeats rapidly in less than a minute, which is a bit confusing.
